### PR TITLE
165100367 Adding in a heartbeat update for the selection database

### PR DESCRIPTION
**Pull request by stormrubio3000**
As the Selection I want to update my database from the ServiceHubs at a regular interval. This will replace having to call and check for updates on every single request.

GIVEN the Selection needs to be updated to make sure its records are in line with those in the ServiceHubs.
WHEN a regular interval of time has passed.
THEN the "HeartBeat" update will make sure that the Selection database is in line with the records from the ServiceHubs.

Dev Notes: As of 4/3/19 the selection calls all the ServiceHubs to check for updates on every call from the UI. This needs to be automated to the regular interval.

### CODE REVIEW

File: src/Housing.Selection.Context/Selection/SelectionService.cs [hunk 1]
```diff
@@ -246,7 +229,8 @@ public async Task RemoveUserFromRoomAsync(AddRemoveUserFromRoomModel model)
 
             // Update in Service hub.
             var apiRoom = _mapper.Map<ApiRoom>(room);
-            await _serviceRoomCalls.UpdateRoomAsync(apiRoom);
+            //TODO: Replace with queue call
```
Comment (c-troha) [inline src/Housing.Selection.Context/Selection/SelectionService.cs]: Remove "TODO" comments, they should not make it to broker

File: src/Housing.Selection.Context/Selection/SelectionService.cs [hunk 2]
```diff
@@ -149,7 +131,8 @@ public async Task AddUserToRoomAsync(AddRemoveUserFromRoomModel model)
 
             //Update the service hub room with changes.
             var apiRoom = _mapper.Map<ApiRoom>(room);
-            await _serviceRoomCalls.UpdateRoomAsync(apiRoom);
+            //TODO: Update implementation with queue
```
Comment (c-troha) [inline src/Housing.Selection.Context/Selection/SelectionService.cs]: Get rid of this comment

File: src/Housing.Selection.Context/Housing.Selection.Context.csproj [hunk 3]
```diff
@@ -5,6 +5,10 @@
     <LangVersion>latest</LangVersion>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+  </PropertyGroup>
```
Comment (escalonn) [inline src/Housing.Selection.Context/Housing.Selection.Context.csproj]: why are we adding this

File: src/Housing.Selection.Context/Selection/ISelectionService.cs [hunk 4]
```diff
@@ -49,5 +49,12 @@ public interface ISelectionService
         /// </summary>
         /// <param name="addRemoveUserFromRoomModel">UserId and RoomId for removing user matching Id to room matching Id.</param>
         Task RemoveUserFromRoomAsync(AddRemoveUserFromRoomModel model);
+        /// <summary>
+        /// Updates local DB with the contents of all hubs (checked periodically)
+        /// </summary>
+        /// <returns>An empty task</returns>
+        Task UpdateFromServiceHub();
+
+
```
Comment (escalonn) [inline src/Housing.Selection.Context/Selection/ISelectionService.cs]: empty line between methods; no extra empty lines here

File: src/Housing.Selection.Context/Selection/SelectionService.cs [hunk 5]
```diff
@@ -39,13 +42,17 @@ public class SelectionService : ISelectionService
         ///<value>Mapper object.</value>
         private readonly IMapper _mapper;
 
+        private ILogger<SelectionService> _logger;
```
Comment (escalonn) [inline src/Housing.Selection.Context/Selection/SelectionService.cs]: let's make this readonly, and non-generic type. (the ctor param should remain generic)

File: src/Housing.Selection.Context/Selection/ISelectionService.cs (hunk 4 above)
Comment (escalonn) [reply src/Housing.Selection.Context/Selection/ISelectionService.cs]: well i guess if every interface in this sln is not having that empty line then we can keep it consistent here

File: src/Housing.Selection.Context/Housing.Selection.Context.csproj [hunk 6]
```diff
@@ -5,6 +5,7 @@
     <LangVersion>latest</LangVersion>
   </PropertyGroup>
 
+
```
Comment (escalonn) [inline src/Housing.Selection.Context/Housing.Selection.Context.csproj]: 🤔

File: src/Housing.Selection.Context/Selection/ISelectionService.cs (hunk 4 above)
Comment (escalonn) [reply src/Housing.Selection.Context/Selection/ISelectionService.cs]: delete the extra empty lines though

File: src/Housing.Selection.Library/Housing.Selection.Library.csproj [hunk 7]
```diff
@@ -5,6 +5,10 @@
     <LangVersion>latest</LangVersion>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+  </PropertyGroup>
+
```
Comment (escalonn) [inline src/Housing.Selection.Library/Housing.Selection.Library.csproj]: still here in this & the other projects too

**Pull request merged**